aten/src/ATen/core/ivalue_inl.h

```diff
@@ -390,6 +399,21 @@ struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
   std::vector<IValue> slots_;
 };
 
+// ivalue Holder that hold a PyObject*
+struct C10_EXPORT ivalue::PythonObject final : c10::intrusive_ptr_target {
+ public:
+  static c10::intrusive_ptr<PythonObject> create(PyObject* py_obj) {
+    return c10::make_intrusive<PythonObject>(py_obj);
+  }
+  PyObject* getPyObject() {
+    return py_obj_;
+  }
+
+  PythonObject(PyObject* py_obj): py_obj_(py_obj) {}
+ private:
+    PyObject* py_obj_;
```

Comment:
I don't see ref-counting logic for this, meaning this is either a dangling free or a leak of the PyObject.

Reply:
hmm yeah I was mimicing `ivalue::Object` and I did not see any ref-counting logic for it either... which is weird. I will update with ref-counting logic.

Reply:
This is missing a decref on the PyObject in the destructor of the holder, which will cause this to leak any objects it holds. This will require having PyObjectHolder be abstract in this file and have concrete instances of it be implemented by a subclass in libtorch_python.